### PR TITLE
feat: support query for influxdb in bindings

### PR DESCRIPTION
**Pull request by uptutu**
# Description

support query for influxdb in bindings

### CODE REVIEW

**Comment by daixiang0** [comment]
Please add tests.

**Comment by uptutu** [comment]
> Please add tests.

Hi @daixiang0 Would you like to take another look? I added the test.

File: bindings/influx/influx.go [hunk 1]
```diff
@@ -16,21 +16,39 @@ package influx
...
 
 	influxdb2 "github.com/influxdata/influxdb-client-go"
 	"github.com/influxdata/influxdb-client-go/api"
+	"github.com/pkg/errors"
 
 	"github.com/dapr/components-contrib/bindings"
 	"github.com/dapr/kit/logger"
 )
 
+const queryOperation bindings.OperationKind = "query"
```
Comment (daixiang0) [inline bindings/influx/influx.go:29]: please combine const field.

File: bindings/influx/influx.go [hunk 2]
```diff
@@ -16,21 +16,39 @@ package influx
...
+
+var (
+	ErrInvalidRequestData      = errors.New("Influx Error: Cannot convert request data")
+	ErrCannotWriteRecord       = errors.New("Influx Error: Cannot write point")
+	ErrInvalidRequestOperation = errors.Errorf("invalid operation type. Expected %s or %s", queryOperation, bindings.CreateOperation)
+	ErrMetadataMissing         = errors.New("metadata required")
+	ErrMetadataRawNotFound     = errors.Errorf("required metadata not set: %s", rawQueryKey)
+)
+
+var _ bindings.OutputBinding = &Influx{}
```
Comment (daixiang0) [inline bindings/influx/influx.go:44]: Same as above.

File: bindings/influx/influx.go [hunk 3]
```diff
@@ -95,31 +114,60 @@ func (i *Influx) getInfluxMetadata(metadata bindings.Metadata) (*influxMetadata,
...
+		err = i.writeAPI.WriteRecord(context.Background(), line)
+		if err != nil {
+			return nil, ErrCannotWriteRecord
+		}
+		return nil, nil
+	case queryOperation:
+		if req.Metadata == nil {
+			return nil, ErrMetadataMissing
+		}
+		i.logger.Debugf("operation: %v", req.Operation)
```
Comment (daixiang0) [inline bindings/influx/influx.go:142]: Please remove.

File: bindings/influx/influx.go [hunk 4]
```diff
@@ -95,31 +114,60 @@ func (i *Influx) getInfluxMetadata(metadata bindings.Metadata) (*influxMetadata,
...
+		return resp, nil
+	default:
+		i.logger.Warnf("unsupported operation invoked")
+		return nil, ErrInvalidRequestOperation
 	}
-
-	line := fmt.Sprintf("%s,%s %s", jsonPoint["measurement"], jsonPoint["tags"], jsonPoint["values"])
-
-	// write the point
-	err = i.writeAPI.WriteRecord(context.Background(), line)
```
Comment (daixiang0) [inline bindings/influx/influx.go:112]: Why remove this?

File: bindings/influx/influx.go (hunk 1 above)
Comment (uptutu) [reply bindings/influx/influx.go:29]: I think the groups of what are not similar in type although they are the same declared variable, so it is intentional.

Same as L44. This is used to check whether a structure implements an interface
```go
var _ bindings.OutputBinding = &Influx{}
```
ref:[uber-go Guide](https://github.com/uber-go/guide/blob/master/style.md#group-similar-declarations)

File: bindings/influx/influx.go (hunk 4 above)
Comment (uptutu) [reply bindings/influx/influx.go:112]: I didn't remove the function there, I just moved it to `case bindings.CreateOperation`

File: bindings/influx/influx.go (hunk 3 above)
Comment (uptutu) [reply bindings/influx/influx.go]: done.

**Comment by yaron2** [comment]
@uptutu please see the linter failures: https://github.com/dapr/components-contrib/runs/4931174929?check_suite_focus=true.

**Comment by yaron2** [comment]
ping @uptutu

**Comment by uptutu** [comment]
I executed `gofumpt -l -w` on the code. I dont know it is ok to pass the github action ?

**Comment by yaron2** [comment]
@uptutu this is merged! thanks so much for the PR and apologies it has taken so long to merge. This need a documentation update, can you please open a PR in the docs repo with the query details?

**Comment by msfussell** [comment]
@uptutu - Can you please add the documentation for this issue https://github.com/dapr/docs/issues/2321
This is required.

**Comment by msfussell** [comment]
@uptutu - Following up here on documentation. Can you update the docs for this change?

**Pull request merged**